service/resource/endpoint/current.go

```diff
@@ -0,0 +1,50 @@
+package endpoint
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework/context/canceledcontext"
+
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+
+	pod, err := toPod(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	podAnnotations := pod.GetAnnotations()
+	podIP, ipOk := podAnnotations["endpoint.kvm.giantswarm.io/ip"]
+	podServiceName, serviceNameOk := pod.Annotations["endpoint.kvm.giantswarm.io/service"]
+
+	if !(ipOk && serviceNameOk) {
+		canceledcontext.SetCanceled(ctx)
+		if canceledcontext.IsCanceled(ctx) {
+			r.logger.Log("pod", pod.GetName(), "debug", "canceling reconciliation for pod")
+
+			return nil, nil
+		}
+	}
+
+	endpoint := Endpoint{
+		ServiceName:      podServiceName,
+		ServiceNamespace: pod.GetNamespace(),
+		EndpointAddress: apiv1.EndpointAddress{
```

Comment:
I used this here with the thought of easily modifying and replacing the actual `EndpointAddress`, as an Endpoint has a List of these Addresses. So it could be (hopefully) be easy to check if the Endpoint contains is.

Reply:
I am not sure what the best way to do this would be. I would maybe just take the magic from the endpoint-updater as it is and do it the same way. An alternative could maybe be to have an endpoint structure like you have here and apply a patch against the real endpoint to merge in the new IP. When we would only update this we would always replace the existing endpoint IPs with the single one we have found here.

Reply:
This was already basically the same in k8s-endpoint-updater. The idea is not to replace completely but instead to add/merge/delete a single address our of the list of addresses in an endpoint. I did change the `apiv1.EndpointAddress` to a `string` with just the IP as it's simpler.

Reply:
Ok, I just thought tracking the IP without the other structure garbage would be good enough. Up to you.